src/lib/spinel/radio_spinel.hpp

```diff
@@ -134,6 +134,28 @@ struct RadioSpinelCallbacks
      */
     void (*mDiagTransmitDone)(otInstance *aInstance, otRadioFrame *aFrame, Error aError);
 #endif // OPENTHREAD_CONFIG_DIAG_ENABLE
+
+    /**
+     * This method writes the radio spinel metrics to the temporary storage.
+     *
+     * @param[in]  aMetrics   A reference to the radio spinel metrics.
+     * @param[in]  aContext   A pointer to application-specific context.
+     */
+    void (*mWriteRadioSpinelMetrics)(const otRadioSpinelMetrics &aMetrics, void *aContext);
```

Comment:
Maybe same here? Have `Save` and `Restore` callbacks? What do you think?

Reply:
The `Save` and `Restore` should be better. I have updated it.